extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -0,0 +1,226 @@
+      );
+    }
+
+    prompt.classList.add('i-amp-story-quiz-prompt');
+
+    const options = toArray(this.element.querySelectorAll('option'));
+    if (options.length < 2 || options.length > 4) {
+      user().error(TAG, 'Improper number of options');
+    }
+
+    this.shadowRoot_
+      .querySelector('.i-amp-story-quiz-prompt-container')
+      .appendChild(prompt);
+
+    options.forEach((option, index) => this.configureOption_(option, index));
+
+    if (this.element.children.length !== 0) {
+      user().error(TAG, 'Too many children');
+    }
+
+    if (this.shadowRoot_ !== this.element.shadowRoot) {
+      this.element.append(this.shadowRoot_);
+    }
+  }
+
+  /**
+   * Creates an option container with option content,
+   * adds styling and answer choices,
+   * and adds it to the shadow DOM.
+   * @param {HTMLOptionElement} option
+   * @param {number} index
+   * @private
+   */
+  configureOption_(option, index) {
+    const html = htmlFor(option);
+    const convertedOption = html`
+      <span class="i-amp-story-quiz-option">
+        <span class="i-amp-story-quiz-answer-choice"></span>
+      </span>
+    `;
+
+    // Fill in the answer choice
+    convertedOption.querySelector(
+      '.i-amp-story-quiz-answer-choice'
+    ).textContent = answerChoiceOptions[index];
+
+    // Transfer the option information into a span then remove the option
+    const optionText = document.createTextNode(option.textContent);
+    convertedOption.append(optionText);
+
+    if (option.hasAttribute('correct')) {
+      convertedOption.setAttribute('correct', 'correct');
+    }
+    option.remove();
+
+    // Add the option to the shadow root
+    this.shadowRoot_
+      .querySelector('.i-amp-story-quiz-option-container')
+      .appendChild(convertedOption);
+  }
+
+  /**
+   * Attaches functions to each option to handle state transition.
+   * @private
+   */
+  attachOptionActionHandlers_() {
+    const options = toArray(
+      this.shadowRoot_.querySelectorAll('.i-amp-story-quiz-option')
+    );
+
+    // Add a click listener to the element to handle option selection and navigation via tapping the prompt
+    this.element.addEventListener('click', e => {
+      let resolved = false;
+      e.composedPath().forEach(p => {
+        if (!resolved) {
+          // An option was selected
+          if (p.getAttribute('class').includes('i-amp-story-quiz-option')) {
+            if (!this.hasReceivedResponse_) {
+              p.classList.add('i-amp-story-quiz-option-selected');
+              options.forEach(o => {
+                o.classList.add('i-amp-story-quiz-option-post-selection');
+
+                const symbolContainer = o.querySelector(
+                  '.i-amp-story-quiz-answer-choice'
+                );
+
+                // TODO(jackbsteinberg): Replace text with icons when the assets are ready
+                if (o.hasAttribute('correct')) {
+                  symbolContainer.textContent = '✓';
```

Comment:
Re: setting some kind of class/attribute on the option, and handling the styling with CSS

Reply:
Done